### ISSUE TITLE
2.8.19 when program starts it stucks on Searching for movies

### ISSUE THREAD

**Issue by nkatsidis** [Komet/MediaElch#1489]
Windows 10

2.8.18 works as intended...

2.8.19 when the program starts it displays Searching for Movies and then it takes for ever to start.

Movies are on network server that runs win10 also

**Comment by bugwelle** [comment]
Ok. There's no difference between those version code-wise. Seems to be a race condition, I think...

TODO (for me):

- Build with Qt 6.4 and ThreadSanitizer enabled (only Qt 6.4 supports TSan)

**Comment by ironhussar** [comment]
I have this issue on 2.8.18 - Windows 10, with media files on a NAS. I have 4 media drives, and sometimes the process gets stuck on disk 1, sometimes on disk 2 etc. Once it happens, even if I kill the task and try again, it won't scan properly. Sometimes works after a reboot, sometimes not.

I went back to 2.8.16 and everything works perfectly again.

**Comment by bugwelle** [comment]
Note to self: TSan with Qt 6.4 requires custom glib: https://forum.qt.io/topic/140544/qt-6-4-built-with-tsan-but-how-to-add-my-own-glib-build/2

**Comment by bugwelle** [comment]
Did you click on "reload" or does this happen on startup (assuming no directory is marked "auto reload")?

**Comment by ironhussar** [comment]
For me it was on startup (no auto-reloading). Still using 2.8.16, which works flawlessly each time.

**Comment by ironhussar** [comment]
I have 4 movie locations (that is, 4 separate drives in a single NAS) and 2 tv locations, by default I have the program start into the Movies section.

I don't keep disc structures on these drives, just the main titles themselves.

**Comment by bugwelle** [comment]
Reproduced on my Windows system

<img width="243" alt="Unbenannt" src="https://user-images.githubusercontent.com/1667306/208308338-e4bc7362-35f2-4a04-9b53-d4a8a48784c7.PNG">

**Comment by bugwelle** [comment]
I didn't get to debug this issue but am able to reproduce the issue (every second to fifth start). I will try to look into it tomorrow or Tuesday, otherwise in the new year (due to vacation, I'm not available between Christmas and New Year's Eve). :)

**Comment by bugwelle** [comment]
Easily reproducible on all system with this patch applied:

```dif
diff --git a/src/file_search/movie/MovieFileSearcher.cpp b/src/file_search/movie/MovieFileSearcher.cpp
index c2390e40..f49549ae 100644
--- a/src/file_search/movie/MovieFileSearcher.cpp
+++ b/src/file_search/movie/MovieFileSearcher.cpp
@@ -91,6 +91,7 @@ void MovieFileSearcher::onDirectoryLoaded(MovieLoader* job)
     Q_ASSERT(job == m_currentJob);
     m_currentJob = nullptr;
     job->deleteLater();
+    QThread::sleep(2);
 
     if (m_aborted || job->isAborted()) {
         // To avoid changes to the model, _after_ the users aborts, don't add any
```

Found the root cause. Will be fixed ASAP.

**Comment by bugwelle** [comment]
Hi, I've found the issue and fixed it in #1510. The next Nightly version will contain the fix. Thank you for reporting this bug!
